lib/carousel_slider.dart

```diff
@@ -37,24 +37,26 @@ class CarouselSlider extends StatefulWidget {
   CarouselSlider(
       {required this.items,
       required this.options,
-      carouselController,
+      CarouselController? carouselController,
       Key? key})
       : itemBuilder = null,
         itemCount = items != null ? items.length : 0,
-        _carouselController = carouselController ??
-            CarouselController() as CarouselControllerImpl,
+        _carouselController = carouselController != null
+            ? carouselController as CarouselControllerImpl
+            : CarouselController() as CarouselControllerImpl,
         super(key: key);
 
   /// The on demand item builder constructor
   CarouselSlider.builder(
       {required this.itemCount,
       required this.itemBuilder,
       required this.options,
-      carouselController,
```

Comment:
This make `carouselController` of type dynamic means any type of parameter can be passed.